### ISSUE TITLE
Aggregate url and username are not saved.

### ISSUE THREAD

**Issue by kkrawczyk123** [opendatakit/briefcase#475]
#### Software versions 
Briefcase v1.10.x, Java v1.8.x, operating system Ubuntu, MacOS, Windows

#### Problem description
Aggregate's url and username used to be saved when connecting to the server on pull and push tabs and when checkbox on settings tab selected - password was stored too. Save password doesn't work right now too - is it needed only to pull before export? 
Do we need this reset button? Only Configure could be enough. Reset button fit more on Aggregate Server Connection window and give a user choice if he wants to reset credentials not to force him to do it every time? 
The best solution would be to store aggregate url as a list as collect does:
![screenshot_20180528-144916](https://user-images.githubusercontent.com/33342488/40618101-bacda482-6290-11e8-95c7-487d0a9f38d7.png)

What do you think @yanokwa @lognaturel and @ggalmazor ?

**Comment by ggalmazor** [comment]
Hi, @kkrawczyk123!

Now I realize that the behavior has changes a little bit. Now Briefcase will remember the connection settings and when you close Briefcase and launch it again, you'll get automatically connected to the server and the forms will be pulled.

It works the same way in the Push tab.

Before this PR, Briefcase remembered only the connection settings of the last server you introduced, which is the same that does now, except you don't see the fields filled for you in any dialog, because you don't need to "configure" it again.

I agree that a new feature that remembers all the configurations you enter would be a great addition.

I also think that a "refresh forms" button for the pull tab could be useful too.

**Comment by ggalmazor** [comment]
@opendatakit-bot label "needs discussion"

**Comment by yanokwa** [comment]
Agreed that this would be a nice to have, but it doesn't feel like a priority just yet.
